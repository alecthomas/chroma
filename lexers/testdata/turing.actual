var x : int := 10

fcn testFunction(a, b, c)
    result (a + b) * c
end testFunction

put testFunction(x, 2 * x, 3)