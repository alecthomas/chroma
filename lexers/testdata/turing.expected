[
  {"type":"KeywordDeclaration","value":"var"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"x"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":":"},
  {"type":"Text","value":" "},
  {"type":"KeywordType","value":"int"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":":="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"10"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"fcn"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"testFunction"},
  {"type":"Punctuation","value":"("},
  {"type":"NameOther","value":"a"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"b"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"c"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"result"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"("},
  {"type":"NameOther","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"b"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"*"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"c"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordDeclaration","value":"end"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"testFunction"},
  {"type":"Text","value":"\n\n"},
  {"type":"Keyword","value":"put"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"testFunction"},
  {"type":"Punctuation","value":"("},
  {"type":"NameOther","value":"x"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"2"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"*"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"x"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"3"},
  {"type":"Punctuation","value":")"}
]
